api/v1alpha1/groupsync_types.go

```diff
@@ -29,6 +29,7 @@ const (
 )
 
 // GroupSyncSpec defines the desired state of GroupSync
+// +k8s:openapi-gen=true
```

Comment:
why did you need this and all the rest linked to creating the openapi?

Reply:
its needed to support openapi generation. otherwise the resources do not get generated by `openapi-gen`

Reply:
I understand that, the question was why do you need to do openapi
generation?

Reply:
Its being added for potential future needs for completeness